From 9139706956d3867a81e91222aac7e3c4f9326d55 Mon Sep 17 00:00:00 2001
From: Juan Gutierrez <juan.gutierrez@nxp.com>
Date: Mon, 24 Apr 2017 16:17:28 -0500
Subject: [PATCH 1728/5242] MXSCM-293-2 arm: dts: add oob and modem reset
 support for SCM devices

commit  042c8419f07f1477ff81003b4903195e1d4123c4 from
https://source.codeaurora.org/external/imx/linux-imx.git

- Add BT modem device reset node for SCM i.MX6 devices
- Add Wifi OOB support for QWKS rev3 and EVB

Signed-off-by: Juan Gutierrez <juan.gutierrez@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6dqscm-qwks-rev3-btwifi.dtsi |    9 +++++++++
 arch/arm/boot/dts/imx6sxscm-evb-btwifi.dtsi       |   10 ++++++++++
 2 files changed, 19 insertions(+)

diff --git a/arch/arm/boot/dts/imx6dqscm-qwks-rev3-btwifi.dtsi b/arch/arm/boot/dts/imx6dqscm-qwks-rev3-btwifi.dtsi
index 3b6f080..dc41db6 100644
--- a/arch/arm/boot/dts/imx6dqscm-qwks-rev3-btwifi.dtsi
+++ b/arch/arm/boot/dts/imx6dqscm-qwks-rev3-btwifi.dtsi
@@ -7,6 +7,13 @@
  */
 
 / {
+	modem_reset: modem-reset {
+		compatible = "gpio-reset";
+		reset-gpios = <&gpio6 1 GPIO_ACTIVE_LOW>;
+		reset-delay-us = <1000>;
+		#reset-cells = <0>;
+	};
+
 	regulators {
 		wlreg_on: fixedregulator@100 {
 			compatible = "regulator-fixed";
@@ -22,6 +29,7 @@
 	bcmdhd_wlan_0: bcmdhd_wlan@0 {
 		compatible = "android,bcmdhd_wlan";
 		wlreg_on-supply = <&wlreg_on>;
+		gpios = <&gpio6 4 0>; /* WL_HOST_WAKE */
 	};
 };
 
@@ -30,6 +38,7 @@
 	pinctrl-0 = <&pinctrl_uart4
 		     &pinctrl_bt>;
 	fsl,uart-has-rtscts;
+	resets = <&modem_reset>;
 	status = "okay";
 };
 
diff --git a/arch/arm/boot/dts/imx6sxscm-evb-btwifi.dtsi b/arch/arm/boot/dts/imx6sxscm-evb-btwifi.dtsi
index f55ac0d..3658389 100644
--- a/arch/arm/boot/dts/imx6sxscm-evb-btwifi.dtsi
+++ b/arch/arm/boot/dts/imx6sxscm-evb-btwifi.dtsi
@@ -7,6 +7,13 @@
  */
 
 / {
+	modem_reset: modem-reset {
+		compatible = "gpio-reset";
+		reset-gpios = <&gpio4 9 GPIO_ACTIVE_LOW>;
+		reset-delay-us = <1000>;
+		#reset-cells = <0>;
+	};
+
 	regulators {
 		wlreg_on: fixedregulator@100 {
 			compatible = "regulator-fixed";
@@ -26,6 +33,7 @@
 	bcmdhd_wlan_0: bcmdhd_wlan@0 {
 		compatible = "android,bcmdhd_wlan";
 		wlreg_on-supply = <&wlreg_on>;
+		gpios = <&gpio7 9 0>; /* WL_HOST_WAKE */
 	};
 };
 
@@ -90,6 +98,7 @@
 				MX6SX_PAD_SD2_DATA3__USDHC2_DATA3	0x17069
 				/* Murata Module control signals */
 				MX6SX_PAD_NAND_DATA04__GPIO4_IO_8	0x13069
+				MX6SX_PAD_SD3_DATA7__GPIO7_IO_9         0x13069
 			>;
 		};
 	};
@@ -100,6 +109,7 @@
 	pinctrl-0 = <&pinctrl_uart6
 		     &pinctrl_bt>;
 	fsl,uart-has-rtscts;
+	resets = <&modem_reset>;
 	status = "okay";
 };
 
-- 
1.7.9.5

